.github/workflows/index-resources.yml

```diff
@@ -21,6 +28,8 @@ jobs:
         shell: bash
     steps:
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || github.sha }}
```

Comment:
Why in a `workflow_call` invocation is `github.sha` not what we want (it should be the `github.sha` of the caller¹)? i.e. why do we need `inputs.ref`? i.e. why would we want to run CI for ref A but reindex with ref B?

¹ "When a reusable workflow is triggered by a caller workflow, the `github` context is always associated with the caller workflow." (https://docs.github.com/en/actions/using-workflows/reusing-workflows)

Reply:
Ah, yeah looking at this now, this was unnecessary for this particular commit. It's clearer in https://github.com/nextstrain/nextstrain.org/pull/841/commits/238db45f13995856cef8701cd5b530cbcaff1da7 where I've conditionalized jobs on the presence of the `input.ref` rather than a list of workflow events.

Reply:
Right, I saw that once I got further. The current conditionals are like:

```yaml
if: ${{ inputs.ref == '' }}
```

and I think that's quite a bit less clear than:

```yaml
if: github.event_name == 'schedule'
```

Reply:
Ah, I wasn't going to use 
```yaml
if: github.event_name == 'schedule'
```
because sometimes we want to manually trigger index rebuilds via `workflow_dispatch`. 

I think checking `input.ref` is able to catch all cases where we don't want to run the default production/canary updates.

Reply:
Hmm. I think it's fairly confusing/tangled. AIUI, `input.ref` could be provided two places:

1. `workflow_call` (i.e. invoked by another workflow) where a `ref` value was passed. The only invoking workflow is CI, which passes `ref: ${{ github.sha }}`.

2. `workflow_dispatch` (i.e. manual invocation) where the `input.ref` textbox is filled in. Note that this is **different** than choosing the ref at which to run the workflow, which is an implicit input that's required. When `input.ref` is provided, it should ~always match the implicit ref input, but nothing guarantees that.

It didn't initially seem to me we'd ever want to do (2), but [you proposed a use case](https://github.com/nextstrain/nextstrain.org/pull/841#discussion_r1588355716).

So it seems like the spelled-out logic would be something like:

- If `github.event_name == 'schedule'`, then update for our two production Heroku apps.
- If `github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/master'`, then update for our two production Heroku apps.
- If `github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/master'`, then update for `github.sha`.
- If `github.event_name == 'workflow_call'`, then update for `github.sha`.

Reply:
Spelled-out the cases to use `GITHUB_SHA` in https://github.com/nextstrain/nextstrain.org/pull/841/commits/a90c1c35da5265a1680ae513bcb701615feaba94.